lib/bamboo/template.ex

```diff
@@ -0,0 +1,271 @@
+
+      defmodule MyApp.Email do
+        use Bamboo.Template, view: MyApp.Email.AccountView
+
+        def welcome_email do
+          new_email()
+          |> put_text_layout({MyApp.Email.LayoutView, "email.text"})
+          |> put_html_layout({MyApp.Email.LayoutView, "email.html"})
+          |> render(:welcome) # Pass atom to render html AND plain text templates
+        end
+      end
+
+  _Set both the text and HTML layout at the same time for an email._
+
+      defmodule MyApp.Email do
+        use Bamboo.Template, view: MyApp.Email.AccountView
+
+        def welcome_email do
+          new_email()
+          |> put_layout({MyApp.Email.LayoutView, :email})
+          |> render(:welcome)
+        end
+      end
+
+  _Render both text and html emails without layouts._
+
+      defmodule MyApp.Email do
+        use Bamboo.Template, view: MyApp.Email.AccountView
+
+        def welcome_email do
+          new_email()
+          |> render(:welcome)
+        end
+      end
+
+  _Make assigns available to a template._
+
+      defmodule MyApp.Email do
+        use Bamboo.Template, view: MyApp.Email.AccountView
+
+        def welcome_email(user) do
+          new_email()
+          |> assign(:user, user)
+          |> render(:welcome)
+        end
+      end
+
+  _Make assigns available to a template during render call._
+
+      defmodule MyApp.Email do
+        use Bamboo.Template, view: MyApp.Email.AccountView
+
+        def welcome_email(user) do
+          new_email()
+          |> put_html_layout({MyApp.Email.LayoutView, "email.html"})
+          |> render(:welcome, user: user)
+        end
+      end
+
+  _Render an email by passing the template string to render._
+
+      defmodule MyApp.Email do
+        use Bamboo.Template, view: MyApp.Email.AccountView
+
+        def html_email do
+          new_email()
+          |> render("html_email.html")
+        end
+
+        def text_email do
+          new_email
+          |> render("text_email.text")
+        end
+      end
+
+  ## HTML Layout Example
+
+      # lib/my_app/email.ex
+      defmodule MyApp.Email do
+        use Bamboo.Template, view: MyApp.Email.AccountView
+
+        def welcome_email(person) do
+          base_email()
+          |> to(person)
+          |> subject("Welcome to MyApp")
+          |> assign(:person, person)
+          |> render(:welcome_email)
+        end
+
+        defp base_email do
+          new_email()
+          |> from("Rob Ot<robot@changelog.com>")
+          |> put_header("Reply-To", "editors@changelog.com")
+          # This will use the "email.html.eex" file as a layout when rendering html emails.
+          # Plain text emails will not use a layout unless you use `put_text_layout`
+          |> put_html_layout({MyApp.Email.LayoutView, "email.html"})
+        end
+      end
+
+      # lib/my_app/email/views/layout_view.ex
+      defmodule MyApp.Email.LayoutView do
+        use Bamboo.View, path: "/lib/my_app/email/templates/layout"
```

Comment:
Just to confirm, should these be absolute paths?

Reply:
Yeah, curious about your thoughts here. I used the full paths to be explicit. I took inspiration from Phoenix, but in Phoenix you define the `root` (I think you can optionally define a partial `path` too). Then Phoenix uses conventions about your View to determine where to find those templates. 

I like that people can just specify whatever path they want here. That's where those templates are and that's what we compile for that particular view.

Reply:
The comment doesn't have the leading slash, but the underlying code does.  Maybe be explicit that this is either "always relative to app root", or do something where if there's a leading slash it's absolute (and without the leading slash, it's relative to app root)?

Reply:
Great observation @dotdotdotpaul. I'm curious, is there a better name for what is currently the key `path` that would help make that clear? I'd like to make it clear in documentation _but also_ make it so it's clear from the usage what you're supposed to set there. Have any thoughts?

Reply:
Could probably add a config option for "root" and then stipulate that "path" is _always_ relative to whatever "root" is defined as, and if "root" is `nil`, it will use the project root (which is probably the preferred default anyway -- I can only see a different root being used if there's some kind of separate template source).  But I've only had one cup of coffee this morning, so that may not be an entirely baked idea.

Reply:
Re-reading these comments with time, I now realize that what I said at first was totally wrong. We don't take absolute paths. @lancejjohnson I thought you were asking something else, and I got confused. I fixed this in 6297d0db381166c6a0b0744e34e86f966fbe6967. 

@dotdotdotpaul I liked your idea. I was going to implement it, but that's when I realized we don't want absolute paths. So `path` here is just relative paths -- relative to the root of the project. And like you say, that's probably the preferred default anyway.